### PR TITLE
Add support for OpenCLIP and LAION models

### PR DESCRIPTION
**Pull request by antas-marcin**

### CODE REVIEW

File: meta.py [hunk 1]
```diff
@@ -9,6 +9,12 @@ def __init__(self):
     if path.exists('./models/openai_clip'):
       # OpenAI CLIP Models
       self._config = CLIPConfig.from_pretrained('./models/openai_clip').to_dict()
+    elif path.exists('./models/openclip'):
+      # OpenCLIP Models
+      config = {}
+      with open(path.join('./models/openclip', "config.json")) as config_file:
+        config = json.load(config_file)
+      self._config = config
```
Comment (StefanBogdan) [inline meta.py:17]: You can make the code shorter:
```python
elif path.exists('./models/openclip'):
      # OpenCLIP Models
      with open(path.join('./models/openclip', "config.json")) as config_file:
              self._config = json.load(config_file)
else:
```

File: download.py [hunk 2]
```diff
@@ -5,9 +5,47 @@
...
+import open_clip
+import json
 
 logging.basicConfig(level=logging.INFO)
 
+open_clip_model_name = os.getenv('OPEN_CLIP_MODEL_NAME')
+open_clip_pretrained = os.getenv('OPEN_CLIP_PRETRAINED')
+
+if open_clip_model_name != "" and open_clip_pretrained != "":
+  def check_model_and_pretrained(model_name: str, pretrained: str):
```
Comment (StefanBogdan) [inline download.py:17]: You can make this function a bit shorter and faster:
```python
def check_model_and_pretrained(model_name: str, pretrained: str):
   if (model_name, pretrained) in open_clip.list_pretrained():
      return
   logging.error("Fatal: Available pairs are:")
   for pair in open_clip.list_pretrained():
      logging.error(f"Fatal: model: {pair[0]} pretrained: {pair[1]}")
   logging.error(f"Fatal: Match not found for OPEN_CLIP model {model_name} with pretrained {pretrained} pair")
   sys.exit(1)
```

File: clip.py [hunk 3]
```diff
@@ -153,14 +156,82 @@ def vectorize(self, payload: ClipInput) -> ClipResult:
...
 
+class ClipInferenceOpenCLIP:
+	clip_model: open_clip.CLIP | open_clip.CoCa | open_clip.CustomTextCLIP
+
+	def __init__(self, cuda, cuda_core):
+		self.device = 'cpu'
+		if cuda:
+			self.device=cuda_core
+
+		config = {}
```
Comment (StefanBogdan) [inline clip.py:167]: You do not need to declare this variable, it will be created and populated in the `with` block below.

**Review by StefanBogdan — COMMENTED** [review]
Seems fine to be, I added some code optimizations. If you want you can add them or you can skip them.

File: clip.py (hunk 3 above)
Comment (antas-marcin) [reply clip.py:167]: oh, didn't know that, thanks!

File: download.py (hunk 2 above)
Comment (antas-marcin) [reply download.py:17]: nice! thanks

File: meta.py (hunk 1 above)
Comment (antas-marcin) [reply meta.py:17]: cool!

**Review by StefanBogdan — APPROVED** [review]

**Pull request merged**